src/main/scala/tech/cryptonomic/conseil/tezos/PlatformDiscoveryOperations.scala

```diff
@@ -5,13 +5,21 @@ import slick.ast.FieldSymbol
 import slick.jdbc.PostgresProfile.api._
 import tech.cryptonomic.conseil.tezos.PlatformDiscoveryTypes.DataType.DataType
 import tech.cryptonomic.conseil.tezos.PlatformDiscoveryTypes._
+import tech.cryptonomic.conseil.tezos.QueryProtocolTypes.Query
 import tech.cryptonomic.conseil.tezos.{TezosDatabaseOperations => TezosDb}
 
 import scala.collection.JavaConverters._
 import scala.concurrent.{ExecutionContext, Future}
 
+/**
+  * Trait containing the interface for the QueryProtocol
+  */
+trait QueryProtocolOperations {
 
-object PlatformDiscoveryOperations {
+  def queryWithPredicates(tableName: String, query: Query)(implicit ec: ExecutionContext): Future[List[Map[String, Any]]]
```

Comment:
How come the query logic is in `PlatformDiscoveryOperations`? We thought platform discovery is about serving metadata while the query classes are for representing queries. Maybe the query logic best belongs in `ApiOperations`?

Reply:
you are right, after moving some things to separate methods I moved this also to `ApiOperations`